rfcs/20210119-determinism.md

```diff
@@ -0,0 +1,82 @@
+# RFC: Enabling Determinism in TensorFlow
+  
+| Status        | Proposed                                             |
+:---------------|:-----------------------------------------------------|
+| **Author(s)** | Pankaj Kanwar (Google), Reed Wanderman-Milne (Google)|
+| **Sponsor**   | Sanjoy Das (Google)                                  |
+| **Updated**   | 2021-01-31                                           |
+
+
+## Objective
+Allow users to enable determinism behavior in TensorFlow. This means if the user runs a TensorFlow program multiple times, the model outputs and weights will be the same each time. Determinism will be supported on CPUs and GPUs.
+ 
+To get deterministic behavior, users must do the following:
+
+* Enable determinism using the API proposed in this doc.
+* Use same hardware in every run.
```

Comment:
This could be misleading. You don't have to use exactly the same hardware instance, just the same hardware configuration. So, in the single accelerator case, that might be the same accelerator silicon architecture. I assume that we're not addressing multi-accelerator determinism here. That's another topic.

I suggest: Use the same hardware configuration in every run.